test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/ExpressionTextCacheTest.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class ExpressionTextCacheTest
+    {
+        private readonly ExpressionTextCache _expressionTextCache = new ExpressionTextCache();
+
+        [Theory]
+        [MemberData(nameof(MemberAccessExpressions))]
+        public void GetExpressionText_CachesMemberAccessExpression(LambdaExpression expression)
+        {
+            // Act - 1
+            var text1 = _expressionTextCache.Get(expression);
+
+            // Assert - 1
+            Assert.NotEmpty(text1);
+
+            // Act - 2
+            var text2 = _expressionTextCache.Get(expression);
+
+            // Assert - 2
+            Assert.NotEmpty(text2);
+            Assert.Same(text1, text2); // Cached
+        }
+
+        [Theory]
+        [MemberData(nameof(IndexerExpressions))]
+        public void GetExpressionText_DoesNotCacheIndexerExpression(LambdaExpression expression)
+        {
+            // Act - 1
+            var text1 = _expressionTextCache.Get(expression);
+
+            // Assert - 1
+            Assert.NotEmpty(text1);
+
+            // Act - 2
+            var text2 = _expressionTextCache.Get(expression);
+
+            // Assert - 2
+            Assert.NotEmpty(text2);
+            Assert.NotSame(text1, text2); // not cached
+        }
+
+        public static IEnumerable<object[]> MemberAccessExpressions
+        {
+            get
+            {
+                var data = new TheoryData<Expression>();
+                data.Add((Expression<Func<TestModel,Category>>)(model => model.SelectedCategory));
+                data.Add((Expression<Func<TestModel, CategoryName>>)(model => model.SelectedCategory.CategoryName));
+                data.Add((Expression<Func<TestModel, int>>)(model => model.SelectedCategory.CategoryId));
+                data.Add((Expression<Func<TestModel, string>>)(model => model.SelectedCategory.CategoryName.MainCategory));
```

Comment:
- Add a test of `var value = "valuable"; ... model => value ...` and use a different lambda parameter e.g. `q` at least once.
- Rather verbose. Why not use dictionary init syntax?